spinalcordtoolbox/scripts/sct_warp_template.py

```diff
@@ -117,14 +118,12 @@ def warp_label(path_label, folder_label, file_label, fname_src, fname_transfo, p
         for i in range(0, len(template_label_file)):
             fname_label = os.path.join(path_label, folder_label, template_label_file[i])
             # apply transfo
-            run_proc('isct_antsApplyTransforms -d 3 -i %s -r %s -t %s -o %s -n %s' %
-                     (fname_label,
-                      fname_src,
-                      fname_transfo,
-                      os.path.join(path_out, folder_label, template_label_file[i]),
-                      get_interp(template_label_file[i], list_labels_nn)),
-                     is_sct_binary=True,
-                     verbose=verbose)
+            sct_apply_transfo.main(['-i', fname_label,
+                                    '-d', fname_src,
+                                    '-w', fname_transfo,
+                                    '-o', os.path.join(path_out, folder_label, template_label_file[i]),
+                                    '-x', get_interp(template_label_file[i], list_labels_nn),
+                                    '-v', str(verbose)])
```

Comment:
This change causes a slowdown from ~4.5m to ~8.5m. 

(Highlighting so that it can be linked to.)

Reply:
@jcohenadad The tl;dr of my investigation in https://github.com/neuropoly/spinalcordtoolbox/issues/3271#issuecomment-806035150 is this:

* `sct_apply_transfo` has some extra logic involving headers, so it loads some `Images` to access the header.
* SCTs' `Image` class loads the data array _immediately_, even if it isn't needed.
    * By comparison, `nibabel.load` waits until the data is explicitly requested (which is much more efficient). 
* This data loading is the source of the extra ~4s per individual call to `sct_apply_transfo`.

Reply:
Excellent investigation, and yes, it makes perfect sense. Thank you for opening this other issue. Now that we know the cause we should easily find a workaround to this increased processing time.